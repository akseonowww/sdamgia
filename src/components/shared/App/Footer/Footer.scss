.Footer {
	font-size: 9pt;
}
