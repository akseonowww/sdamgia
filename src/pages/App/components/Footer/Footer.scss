@import "src/styles/breakpoints";

.Footer {
  display: flex;
  flex-direction: column;

  @include media-breakpoint-up(mil) {
    & {
      flex-direction: row;
      justify-content: space-between;
    }
  }

  &-Menu {
    margin-bottom: 0.5rem;
  }

  &-Copyright {
    margin-bottom: 1rem;
  }

  &-Menu,
  &-Copyright {
    font-size: 1rem;
    line-height: 1.4rem;
  }

  &-Ad {
    order: 3;

    @include media-breakpoint-up(mil) {
      & {
        order: initial;
      }
    }
  }
}
