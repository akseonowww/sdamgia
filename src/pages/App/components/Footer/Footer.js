import React from 'react';

import './Footer.scss';

import spbsut from './assets/spbsut.png';

const Footer = ({ className }) => (
	<footer className={`Footer ${className}`}>
		<p>
			<a class="Link Link_black" href="/about">
				О проекте
			</a>{' '}
			·{' '}
			<a class="Link Link_black" href="/staff">
				Редакция
			</a>{' '}
			·{' '}
			<a class="Link Link_black" href="/legal">
				Правовая информация
			</a>{' '}
		</p>
		<a
			href="http://priem.sut.ru/"
			target="_blank"
			rel="noopener noreferrer"
		>
			<img src={spbsut} alt="СПб ГУТ!" title="priem.sut.ru" width="160" />
		</a>
		<p>©&nbsp;Гущин Д. Д., 2011—2019</p>
	</footer>
);

export default Footer;
