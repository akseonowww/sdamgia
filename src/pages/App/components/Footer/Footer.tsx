import React, { FC } from 'react'
import cx from 'classnames'

import Link from 'components/Link'
import './Footer.scss'

import spbsut from './assets/spbsut.png'

interface IFooterProps {
  className?: string
}

const Footer: FC<IFooterProps> = ({ className }) => (
  <footer className={cx('Footer', className)}>
    <nav className="Footer-Menu">
      <Link className="Link_black" href="/about">
        О проекте
      </Link>{' '}
      ·{' '}
      <Link className="Link_black" href="/staff">
        Редакция
      </Link>{' '}
      ·{' '}
      <Link className="Link_black" href="/legal">
        Правовая&nbsp;информация
      </Link>{' '}
    </nav>
    <a
      className="Footer-Ad"
      href="http://priem.sut.ru/"
      target="_blank"
      rel="noopener noreferrer"
    >
      <img src={spbsut} alt="СПб ГУТ!" title="priem.sut.ru" width="160" />
    </a>
    <div className="Footer-Copyright">
      ©&nbsp;Гущин Д. Д., 2011—{new Date().getFullYear()}
    </div>
  </footer>
)

export default Footer
