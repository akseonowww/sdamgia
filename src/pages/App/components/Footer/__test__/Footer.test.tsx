import React from 'react'
import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import pretty from 'pretty'
import { MemoryRouter } from 'react-router-dom'

import Footer from '../Footer'

describe('Footer', () => {
  let container: HTMLElement | null = null

  beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    if (container) {
      unmountComponentAtNode(container)
      container.remove()
      container = null
    }
  })

  it('renders with or without a className', () => {
    act(() => {
      render(
        <MemoryRouter>
          <Footer />
        </MemoryRouter>,
        container
      )
    })
    expect(container && container.textContent).toBe(
      `О проекте · Редакция · Правовая информация © Гущин Д. Д., 2011—${new Date().getFullYear()}`
    )

    act(() => {
      render(
        <MemoryRouter>
          <Footer className="PageLayout-Footer" />
        </MemoryRouter>,
        container
      )
    })
    expect(container && container.textContent).toBe(
      `О проекте · Редакция · Правовая информация © Гущин Д. Д., 2011—${new Date().getFullYear()}`
    )
  })

  it('should render a content', () => {
    act(() => {
      render(
        <MemoryRouter>
          <Footer />
        </MemoryRouter>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()

    act(() => {
      render(
        <MemoryRouter>
          <Footer className="PageLayout-Footer" />
        </MemoryRouter>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()
  })
})
