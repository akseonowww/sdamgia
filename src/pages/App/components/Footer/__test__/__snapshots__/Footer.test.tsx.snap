// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render a content 1`] = `
"<footer class=\\"Footer\\">
  <nav class=\\"Footer-Menu\\"><a class=\\"Link Link_black\\" href=\\"/about\\" rel=\\"noopener noreferrer\\">О&nbsp;проекте</a> · <a class=\\"Link Link_black\\" href=\\"/staff\\" rel=\\"noopener noreferrer\\">Редакция</a> · <a class=\\"Link Link_black\\" href=\\"/legal\\" rel=\\"noopener noreferrer\\">Правовая&nbsp;информация</a> </nav><a class=\\"Footer-Ad\\" href=\\"http://priem.sut.ru/\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><img src=\\"spbsut.png\\" alt=\\"СПб ГУТ!\\" title=\\"priem.sut.ru\\" width=\\"160\\"></a>
  <div class=\\"Footer-Copyright\\">©&nbsp;Гущин Д.&nbsp;Д., 2011—2020</div>
</footer>"
`;

exports[`Footer should render a content 2`] = `
"<footer class=\\"Footer PageLayout-Footer\\">
  <nav class=\\"Footer-Menu\\"><a class=\\"Link Link_black\\" href=\\"/about\\" rel=\\"noopener noreferrer\\">О&nbsp;проекте</a> · <a class=\\"Link Link_black\\" href=\\"/staff\\" rel=\\"noopener noreferrer\\">Редакция</a> · <a class=\\"Link Link_black\\" href=\\"/legal\\" rel=\\"noopener noreferrer\\">Правовая&nbsp;информация</a> </nav><a class=\\"Footer-Ad\\" href=\\"http://priem.sut.ru/\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><img src=\\"spbsut.png\\" alt=\\"СПб ГУТ!\\" title=\\"priem.sut.ru\\" width=\\"160\\"></a>
  <div class=\\"Footer-Copyright\\">©&nbsp;Гущин Д.&nbsp;Д., 2011—2020</div>
</footer>"
`;
